src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/service/ActivityMappingService.kt

```diff
@@ -54,16 +59,49 @@ class ActivityMappingService(
         ),
         null,
       )
-      log.debug("Mapping created with activityScheduleId = $activityScheduleId, nomisCourseActivityId = $nomisCourseActivityId")
+
+      createMappingRequest.scheduledInstanceMappings.map { request ->
+        ActivityScheduleMapping(
+          scheduledInstanceId = request.scheduledInstanceId,
+          nomisCourseScheduleId = request.nomisCourseScheduleId,
+          mappingType = ActivityScheduleMappingType.valueOf(request.mappingType),
+          activityScheduleId = activityScheduleId,
+        ).also { entity ->
+          activityScheduleMappingRepository.save(entity)
+          telemetryClient.trackEvent(
+            "activity-schedule-mappings-created",
+            mapOf(
+              "activityScheduleId" to activityScheduleId.toString(),
+              "scheduleInstanceId" to entity.scheduledInstanceId.toString(),
+              "nomisCourseScheduleId" to entity.nomisCourseScheduleId.toString(),
+            ),
+            null,
+          )
+        }
+      }
     }
+  }
 
   suspend fun getMappingById(id: Long): ActivityMappingDto =
-    activityMappingRepository.findById(id)
-      ?.let { ActivityMappingDto(it) }
-      ?: throw NotFoundException("Activity schedule id=$id")
+    activityScheduleMappingRepository.findAllByActivityScheduleId(id)
+      .map { ActivityScheduleMappingDto(it) }
+      .let { schedules ->
+        activityMappingRepository.findById(id)
+          ?.let { ActivityMappingDto(it, schedules.toList()) }
+          ?: throw NotFoundException("Activity schedule id=$id")
+      }
```

Comment:
We're manually managing the relationship between activity mappings and schedule mappings because Spring Data JDBC wouldn't add much - at least for now.